web-bundle/src/main/java/com/graphhopper/replica/ReplicaCustomSpeedsCarAccessParser.java

```diff
@@ -0,0 +1,28 @@
+package com.graphhopper.replica;
+
+import com.google.common.collect.ImmutableMap;
+import com.graphhopper.reader.ReaderWay;
+import com.graphhopper.routing.ev.EdgeIntAccess;
+import com.graphhopper.routing.ev.EncodedValueLookup;
+import com.graphhopper.routing.util.parsers.CarAccessParser;
+import com.graphhopper.util.PMap;
+import org.apache.commons.lang3.tuple.Pair;
+
+import static com.graphhopper.customspeeds.CustomSpeedsUtils.validateCustomSpeedDirection;
+
+public class ReplicaCustomSpeedsCarAccessParser extends CarAccessParser {
+    private final ImmutableMap<Pair<Long, Boolean>, Double> osmWayIdAndBwdToMaxSpeed;
+    private final boolean directionalCustomSpeedsProvided;
+
+    public ReplicaCustomSpeedsCarAccessParser(EncodedValueLookup lookup, PMap properties, ImmutableMap<Pair<Long, Boolean>, Double> osmWayIdAndBwdToMaxSpeed, boolean directionalCustomSpeedsProvided) {
+        super(lookup, properties);
+        this.osmWayIdAndBwdToMaxSpeed = osmWayIdAndBwdToMaxSpeed;
+        this.directionalCustomSpeedsProvided = directionalCustomSpeedsProvided;
+    }
+
+    @Override
+    public void handleWayTags(int edgeId, EdgeIntAccess edgeIntAccess, ReaderWay way) {
+        super.handleWayTags(edgeId, edgeIntAccess, way);
+        validateCustomSpeedDirection(osmWayIdAndBwdToMaxSpeed, directionalCustomSpeedsProvided, way.getId(), accessEnc, edgeId, edgeIntAccess);
```

Comment:
I'd prefer to run these validations earlier on, eg where we run `CustomSpeedsVehicle.validateCustomSpeeds()` in `getCustomSpeedVehiclesByName()`, but I couldn't find a way to do that in a way that gave us access to GraphHopper's internal notion of directional access for every Way (which is an important part of the check)

Reply:
Good call, clarifying comment added!